File: 0000-0999/0400.NthDigit.py
class Solution(object):
    def findNthDigit(self, n):
        """
        :type n: int
        :rtype: int
        """
        start = 1
        length = 1
        count = 9
        while n > length * count:
            n -= length * count
            start *= 10
            length += 1
            count *= 10
        num = start + (n-1)/length
        return int(str(num)[(n-1)%length])
        

File: 0000-0999/0400.NthDigit.go
import "strconv"

func findNthDigit(n int) int {
	start, length, cnt := 1, 1, 9
	for n > length*cnt {
		n -= length * cnt
		length++
		cnt *= 10
		start *= 10
	}
	s := strconv.Itoa(start + (n-1)/length)
	return int(s[(n-1)%length] - '0')
}